sympy/polys/polytools.py

```diff
@@ -7938,3 +7938,134 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system_poly(polys, **kwargs):
+
+    if not polys:
+        return [[]]
+
+    factored_eqs = []
+    for poly in polys:
+        if not isinstance(poly, Poly):
+            raise TypeError("All inputs must be Poly instances")
+
+        try:
+            constant, factors = factor_list(poly, **kwargs)
+            if not factors and constant.is_zero:
+                return []
+            elif not factors:
+                continue
+            current_factors = []
+            if constant.is_number and constant != 0:
+                current_factors.extend((Poly(factor, poly.gens), []) for factor, _ in factors)
+            else:
+                current_factors.extend((Poly(factor, poly.gens), [Poly(constant, poly.gens)])
+                                       for factor, _ in factors)
+
+            if current_factors:
+                factored_eqs.append(current_factors)
+
+        except ComputationFailed:
+            if kwargs.get('modulus'):
+                raise NotImplementedError("multivariate polynomials over finite fields")
+            raise
```

Comment:
Don't put large blocks of code inside try/except. Here the try/except is not needed anyway since all it does is reraise the exception.